src/QsCompiler/TextProcessor/QsExpressionParsing.fs

```diff
@@ -383,10 +383,11 @@ let private argumentTuple =
 /// Expects tuple brackets around the argument even if the argument consists of a single tuple item. 
 /// Note that this parser has a dependency on the arrayItemExpr, identifier, and tupleItem expr parsers - 
 /// meaning they process the left-most part of the call-like expression and thus need to be evaluated *after* the callLikeExpr parser. 
-let internal callLikeExpr = 
-    attempt ((itemAccessExpr <|> identifier <|> tupledItem expr) .>>. argumentTuple) // identifier needs to come *after* arrayItemExpr
-    |>> fun (callable, arg) -> applyBinary CallLikeExpression () callable arg
-
+let internal callLikeExpr =
+    // identifier needs to come *after* arrayItemExpr
+    itemAccessExpr <|> identifier <|> tupledItem expr .>>. many1 argumentTuple
+    |>> List.Cons
+    |>> List.reduce (applyBinary CallLikeExpression ())
```

Comment:
So I think the reason why this change was simple is because it doesn't support item access directly after a call, like `Foo()[0]` - it needs to be tupled like `(Foo())[0]`. @bettinaheim Do we want to support that without parentheses as well?

Reply:
I believe supporting that wouldn't be *too* difficult - we could do something like `many1 (argumentTuple <|> namedItem <|> arrayIndex)` and reduce that list into a tree.

Reply:
Fundamentally, there are three options: 
Option 1: 
Item access and call combinator have the same precedence and lower precedence than modifiers (functors and unwrap). 
This would allow `Adjoint Unitary()` but the currently supported `Adjoint opArray[i]` would no longer be possible. 
Option 2:
Item access and call combinator have the same precedence and higher precedence than modifiers. 
This is definitively undesirable since then `Adjoint Unitary()` would no longer work.
Option 3: 
Item access and call combinator retain the precedence that they currently have. 
This is the option that is implemented, and I believe the best option. While that disallows to freely combine item access and call combinators (e.g. GetUnitary()[i] is not supported as you point out above), the two scenarios  `Adjoint Unitary()` and  `Adjoint opArray[i]` that I believe to be more important are supported.